subprojects/core/src/integTest/groovy/org/gradle/api/artifacts/dsl/DependencyCollectorDslIntegrationTest.groovy

```diff
@@ -0,0 +1,553 @@
+    }
+
+    String singleString(String group, String name, String version) {
+        return renderString(group + ":" + name + ":" + version)
+    }
+
+    String multiString(String group, String name, String version) {
+        return "module(" + renderString(group) + ", " + renderString(name) + ", " + renderString(version) + ")"
+    }
+
+    String namedArgs(String group, String name, String version) {
+        return "module(" + makeNamedArgs(group: renderString(group), name: renderString(name), version: renderString(version)) + ")"
+    }
+
+    String versionCatalog(String group, String name) {
+        return "libs.$group.$name"
+    }
+
+    String versionCatalogBundle(String bundle) {
+        return "libs.bundles.$bundle"
+    }
+
+    String testFixtures(String expression) {
+        return "testFixtures($expression)"
+    }
+
+    String platform(String expression) {
+        return "platform($expression)"
+    }
+
+    String project() {
+        return "project()"
+    }
+
+    String project(String path) {
+        return "project(" + renderString(path) + ")"
+    }
+
+    String dependencyObject(String group, String name, String version) {
+        return cast("project.dependencies.create(" + singleString(group, name, version) + ")", "ModuleDependency")
+    }
+
+    String files(String path) {
+        return "files(${renderString(path)})"
+    }
+
+    String fileTree(String path) {
+        return "fileTree(${renderString(path)})"
+    }
+
+    String providerOf(String expression) {
+        return "project.provider { $expression }"
+    }
+
+    String newStringBuilder(String expression) {
+        return dsl == GradleDsl.GROOVY ? "new StringBuilder($expression)" : "StringBuilder($expression)"
+    }
+
+    String makeNamedArgs(Map<String, String> expressionMap) {
+        if (dsl == GradleDsl.GROOVY) {
+            return expressionMap.collect { k, v -> "${renderString(k)}: " + v }.join(", ")
+        } else {
+            return expressionMap.collect { k, v -> "$k = " + v }.join(", ")
+        }
+    }
+
+    String declareVariable() {
+        return dsl == GradleDsl.GROOVY ? "def" : "val"
+    }
+
+    String instanceOf() {
+        return dsl == GradleDsl.GROOVY ? "instanceof" : "is"
+    }
+
+    String cast(String variable, String type) {
+        return dsl == GradleDsl.GROOVY ? "((${type}) ${variable})" : "(${variable} as ${type})"
+    }
+
+    String setOf(String expression) {
+        return dsl == GradleDsl.GROOVY ? "[$expression]" : "setOf($expression)"
+    }
+
+    def setup() {
+        createDirs("subproject")
+        settingsFile("""
+            include "subproject"
+
+            rootProject.name = "${PROJECT_NAME}"
+            gradle.rootProject {
+                version = "${PROJECT_VERSION}"
+            }
+        """)
+        versionCatalogFile("""
+        [libraries]
+        ${toTomlIdentifier(FOO_GROUP, FOO_NAME)} = "${FOO_GROUP}:${FOO_NAME}:${FOO_VERSION}"
```

Comment:
💭 I think it would be clearer to just write out the identifiers in the TOML file rather than calculating them.